js/src/common/helpers/listItems.tsx

```diff
@@ -3,29 +3,36 @@ import Component, { ComponentAttrs } from '../Component';
 import Separator from '../components/Separator';
 import classList from '../utils/classList';
 
-export interface ModdedVnodeAttrs {
+type ModdedVnodeAttrs = {
   itemClassName?: string;
   key?: string;
-}
+};
 
-export type ModdedVnode<Attrs> = Mithril.Vnode<ModdedVnodeAttrs, Component<Attrs> | {}> & {
-  itemName?: string;
-  itemClassName?: string;
-  tag: Mithril.Vnode['tag'] & {
-    isListItem?: boolean;
-    isActive?: (attrs: ComponentAttrs) => boolean;
-  };
+type ModdedTag = Mithril.Vnode['tag'] & {
+  isListItem?: boolean;
+  isActive?: (attrs: ComponentAttrs) => boolean;
 };
 
-function isSeparator<Attrs>(item: ModdedVnode<Attrs>): boolean {
-  return item.tag === Separator;
+type ModdedVnode = Mithril.Vnode<ModdedVnodeAttrs> & { itemName?: string; itemClassName?: string; tag: ModdedTag };
+
+type ModdedChild = ModdedVnode | string | number | boolean | null | undefined;
+type ModdedChildArray = ModdedChildren[];
+type ModdedChildren = ModdedChild | ModdedChildArray;
+type ModdedChildrenWithItemName = ModdedChildren & { itemName?: string };
```

Comment:
Why not export this? I understand not exporting most of these, but at least `ModdedVnode` and/or `ModdedChildrenWithItemName`?